_posts/2019-12-19-codeception-4.markdown

```diff
@@ -0,0 +1,72 @@
+---
+layout: post
+title: "Codeception 4.0 Released"
+date: 2019-12-18 01:03:50
+---
+
+We are finally here!
+
+Codeception 4.0 is released and this means a lot for us!
+
+After long months on spitting core into modules, publishing new packages and sertting up build pipelines, we are proud to show you the result.
+
+As you, know - Codeception 4 changes the way it is installed. From now on, Codeception package won't include any modules by default and you will need to install them manually. This can be a bit annoying at first but in the end this will allow you to upgrade each package individually and not relying on the core.
+
+In Codeception 4 we also **added support for Symfony 5**, so if you plan to use the latest Symfony components - you really need to upgrade today.
+
+## Upgrade to Codeception 4
+
+To start upgrading to Codeception 4 bump a version in `composer.json` and run `composer update`.
+
+```
+"codeception/codeception": "^4.0"
+```
+
+To simplify upgrading process **we introduced an upgrade script** which will go through your config file and scan for all included modules. Corresponding packages will be added to composer.json, so you could install all Codeception modules you need. 
+
+Run upgrade script on current project:
+
+```
+php vendor/bin/codecept init upgrade4
+```
+
+Script will upgrade your `composer.json` file and it is recommended to run `composer update` immediately, so new modules will be installed. 
+
+> If you use several Codeception configs for multiple bundles you will need to execute upgrade script for each of them.
+
+Congratulations! Upgrade is done. Now you will be able to use each module individually.
+
+**Thanks to Gintautas Miselis @Naktibalda for splitting the core and working Symfony support**
+
+## What's new in 4.0
+
+Since 4.0 our [longread changelogs](https://codeception.com/changelog) are gone. So to track changes in modules you will need to to to corresponding repositories and read changelogs. For instance, here is [the changelog for Db module](https://github.com/Codeception/module-db/releases).
```

Comment:
> you will need to to to

`to to to` should be `to go to`.

This is a good opportunity to post more frequently to Twitter - because we can announce module releases with changes.